examples_utils/benchmarks/run_benchmarks.py

```diff
@@ -102,44 +103,34 @@ def proc_thread():
     t = threading.Thread(target=proc_thread, name="proc_thread")
     t.start()
 
-    timeouts = [0.5, 1, 2, 3, 6, 15, 30, 60]
-    tcount = 0
-    ts_per_line = 40
     total_time = 0
     timeout_error = False
     while True:
-        # show progress
-        tindex = min(int(tcount / ts_per_line), len(timeouts) - 1)
-        t.join(timeouts[tindex])
-        total_time += timeouts[tindex]
-
-        # Stop monitoring progress when benchmarking process thread dies
+        # Check if benchmarking process thread has terminated every second
+        t.join(1)
         if not t.is_alive():
             break
+        total_time += 1
 
         # Monitor if benchmark has timed out
-        if timeout and total_time >= timeout:
-            logger.critical("TIMEOUT")
+        if (timeout is not None) and (total_time >= timeout):
+            logger.error("TIMEOUT")
             timeout_error = True
             proc.kill()
```

Comment:
Raise an exception?

Reply:
It gets raised later in [line 271](https://github.com/graphcore/examples-utils/blob/d0d395715f8971d16ad43682fa101afe82b642d9/examples_utils/benchmarks/run_benchmarks.py#L271), since a timeout in this function you highlighted interrupts the subprocess, so it returns a non-zero exitcode. This is then picked up by the function that calls this one, and an error is raised